adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenRequestTest.java

```diff
@@ -554,6 +559,306 @@ public void testVerifyBrokerRedirectUriInvalidSignature()
     }
 
 
+    /**
+     * Test for returning a valid stale AT when outage mode is on and the server is down.
+     */
+    @SmallTest
+    public void testResiliencyTokenReturn_outageModeOn_positive() throws PackageManager.NameNotFoundException,
+            NoSuchAlgorithmException, OperationCanceledException, IOException, AuthenticatorException,
+            InterruptedException {
+        // make sure AT's expires_in is expired and ext_expires_in is not expired
+        final Calendar expiredTime = new GregorianCalendar();
+        expiredTime.add(Calendar.MINUTE, -MINUS_MINUITE);
+        final Calendar extendedExpiresTime = new GregorianCalendar();
+        extendedExpiresTime.add(Calendar.MINUTE, EXTEND_MINUS_MINUTE);
+        final ITokenCacheStore cacheStore = getTokenCache(expiredTime.getTime(), false, false, extendedExpiresTime.getTime());
+
+        final FileMockContext mockContext = createMockContext();
+        AuthenticationSettings.INSTANCE.setUseBroker(false);
+        final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                VALID_AUTHORITY, false, cacheStore);
+        //turn on the outage mode
+        authContext.setExtendedLifetimeEnabled(true);
+        //mock http response
+        final HttpURLConnection mockedConnection = Mockito.mock(HttpURLConnection.class);
+        HttpUrlConnectionFactory.mockedConnection = mockedConnection;
+        Util.prepareMockedUrlConnection(mockedConnection);
+        Mockito.when(mockedConnection.getOutputStream()).thenReturn(Mockito.mock(OutputStream.class));
+        Mockito.when(mockedConnection.getInputStream()).thenReturn(Util.createInputStream(Util.getSuccessTokenResponse(false, false)),
+                Util.createInputStream(Util.getSuccessTokenResponse(false, false)));
+        Mockito.when(mockedConnection.getResponseCode()).thenReturn(HttpURLConnection.HTTP_GATEWAY_TIMEOUT, HttpURLConnection.HTTP_GATEWAY_TIMEOUT);
```

Comment:
I  don't think I set the Gateway timeout return a success response. 
Ref: Mockito.when(mockedConnection.getInputStream()).thenReturn(Util.createInputStream(Util.getSuccessTokenResponse(false, false)), Util.createInputStream(Util.getSuccessTokenResponse(false, false)));